CMakeLists.txt

```diff
@@ -48,18 +48,24 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
 add_library(spt3g INTERFACE)
 
 # get version info
-include(Spt3gVersion)
-get_spt3g_version()
-if (SPT3G_VERSION)
-	message(STATUS "Building SPT3G software version ${SPT3G_VERSION}")
-	target_compile_definitions(spt3g INTERFACE -DSPT3G_VERSION="${SPT3G_VERSION}")
+if(NOT BUILD_WHEEL)
+	include(Spt3gVersion)
+	get_spt3g_version()
+	if (SPT3G_VERSION)
+		message(STATUS "Building SPT3G software version ${SPT3G_VERSION}")
+		target_compile_definitions(spt3g INTERFACE -DSPT3G_VERSION="${SPT3G_VERSION}")
+	endif()
 endif()
 
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
 # Boost bits we need
 set(Boost_USE_STATIC_LIBS OFF)
 set(Boost_USE_MULTITHREADED ON)
 set(Boost_USE_STATIC_RUNTIME OFF)
-set(Boost_NO_BOOST_CMAKE ON)
+if(NOT BUILD_WHEEL)
+	set(Boost_NO_BOOST_CMAKE ON)
+endif()
```

Comment:
No, it's because versions of boost after 1.80 or something create a config file by default, and the non-config variant that uses FindBoost is completely deprecated as of cmake 3.30.

Reply:
So shouldn't we switch on boost and/or cmake versions rather than BUILD_WHEEL?

Reply:
Sure, this was just a way to group all of the changes that I made specifically for building the wheels that are otherwise not necessary.  Can switch on cmake versions if you prefer, but can't switch on boost versions, since this is an option that is set before the boost `find_package` is called.

Reply:
Does anything break if you just globally remove this? It was a hack to deal with some decade-old Boost versions as I recall.

Reply:
I would guess not.  See #161 for discussion of more general cmake changes.  We could probably remove all of the special boost handling everywhere.

Reply:
I vote for just deleting it and seeing if anything breaks

Reply:
Yep that's what I went with too.  We use sufficiently new boost libraries with py3-v4 and py3-v5 (and I've updated the clustertools setup script to set CMAKE_PREFIX_PATH properly to detect them), so this should be a non-issue.